src/views/Products/Products.test.jsx

```diff
@@ -0,0 +1,67 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { AppWrapper } from '../../utils/testing';
+import Products from './';
+import categories from '../../mocks/en-us/product-categories.json';
+import products from '../../mocks/en-us/products.json';
+
+const renderComponent = () =>
+  render(
+    <AppWrapper>
+      <Products />
+    </AppWrapper>
+  );
+
+describe('Product Category Sidebar ', () => {
+  beforeEach(() => {
+    renderComponent();
+  });
+
+  it('renders data from API', async () => {
+    const categoryTextList = await screen.findAllByTestId('category-checkbox-text');
+    expect(categoryTextList[0].textContent).toBe(categories.results[0].data.name);
+  });
+
+  it('filters on click', async () => {
+    const categoryIndex = 1;
+    const categoryList = await screen.findAllByTestId('category-checkbox');
+    fireEvent.click(categoryList[categoryIndex]);
+    const productCardCategoryList = await screen.findAllByTestId('product-card-category');
+    expect(productCardCategoryList[0].textContent).toBe(categories.results[categoryIndex].data.name.toLowerCase());
+  });
+});
+
+describe('Product List Pagination ', () => {
+  beforeEach(() => {
+    renderComponent();
+  });
+  it('renders correctly page numbers', async () => {
+    const pageList = await screen.findAllByTestId(/pagination-page/);
+    const lastPage = pageList[pageList.length - 1];
+    expect(parseInt(lastPage.textContent)).toBe(products.total_pages);
```

Comment:
This would be a simpler way:
```suggestion
    const pageList = await screen.findAllByTestId(/pagination-page/);;
    expect(parseInt(pageList.length)).toBe(products.total_pages);
```

Reply:
Sure, this is a simpler way, thanks, but there's something about the pagination component. Depending on the screen size it is not rendering all the pages, instead it renders three dots to not show some pages, so that's way I have to get the text content from the last page.